test/aave-v3/TestSupplyVault.t.sol

```diff
@@ -489,4 +489,130 @@ contract TestSupplyVault is TestSetupVaults {
+            poolTokens,
+            address(daiSupplyVault),
+            rewardToken
+        );
+
+        // Should update the unclaimed amount
+        vaultSupplier2.depositVault(daiSupplyVault, amount, address(vaultSupplier1));
+        (, uint128 userReward1) = daiSupplyVault.userRewards(rewardToken, address(vaultSupplier1));
+        (uint128 index2, uint128 userReward2) = daiSupplyVault.userRewards(
+            rewardToken,
+            address(vaultSupplier2)
+        );
+        assertEq(index2, 0);
+        assertEq(userReward2, 0);
+        assertGt(uint256(userReward1), 0);
+        assertApproxEqAbs(uint256(userReward1), expectedTotalRewardsAmount, 10000);
+    }
+
+    function testRewardsShouldAccrueWhenMintingOnBehalf() public {
+        uint256 amount = 10_000 ether;
+        address[] memory poolTokens = new address[](1);
+        poolTokens[0] = aDai;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+        vm.warp(block.timestamp + 10 days);
+
+        uint256 expectedTotalRewardsAmount = rewardsManager.getUserRewards(
+            poolTokens,
+            address(daiSupplyVault),
+            rewardToken
+        );
+        morpho.updateIndexes(aDai);
+
+        // Should update the unclaimed amount
+        vaultSupplier2.mintVault(
+            daiSupplyVault,
+            daiSupplyVault.previewMint(amount),
+            address(vaultSupplier1)
+        );
+
+        (, uint128 userReward1) = daiSupplyVault.userRewards(rewardToken, address(vaultSupplier1));
+        (uint128 index2, uint128 userReward2) = daiSupplyVault.userRewards(
+            rewardToken,
+            address(vaultSupplier2)
+        );
+        assertEq(index2, 0);
+        assertEq(userReward2, 0);
+        assertGt(uint256(userReward1), 0);
+        assertApproxEqAbs(uint256(userReward1), expectedTotalRewardsAmount, 10000);
+    }
+
+    function testRewardsShouldAccrueWhenRedeemingOnBehalf() public {
+        uint256 amount = 10_000 ether;
+        address[] memory poolTokens = new address[](1);
+        poolTokens[0] = aDai;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+        vm.warp(block.timestamp + 10 days);
+
+        uint256 expectedTotalRewardsAmount = rewardsManager.getUserRewards(
+            poolTokens,
+            address(daiSupplyVault),
+            rewardToken
+        );
+        morpho.updateIndexes(aDai);
+
+        // Should update the unclaimed amount
+        vaultSupplier1.redeemVault(
+            daiSupplyVault,
+            daiSupplyVault.balanceOf(address(vaultSupplier1)),
+            address(vaultSupplier2),
+            address(vaultSupplier1)
+        );
+        (, uint128 userReward1) = daiSupplyVault.userRewards(rewardToken, address(vaultSupplier1));
+        (uint128 index2, uint128 userReward2) = daiSupplyVault.userRewards(
+            rewardToken,
+            address(vaultSupplier2)
+        );
+        assertEq(index2, 0);
+        assertEq(userReward2, 0);
```

Comment:
wouldn't it be interesting to warp forward in time before querying and asserting these?
So that we make sure that vaultSupplier2 doesn't accrue anything (because with the current state of the code, vaultSupplier2 is not entitled to any reward right after an interaction anyway?)